pages/identities.tsx

```diff
@@ -35,31 +37,44 @@ const Identities: NextPage = () => {
     calls: callData,
   });
 
-  function mint() {
-    execute();
-  }
-
-  const { data, error: transactionError } = useWaitForTransaction({
+  const { data: transactionData, isError } = useWaitForTransaction({
     hash: mintData?.transaction_hash,
     watch: true,
   });
 
+  useEffect(() => {
+    if (isError) {
+      setScreen("error");
+    } else if (transactionData?.status === "RECEIVED") {
+      addTransaction({ hash: mintData?.transaction_hash ?? "" });
+      setScreen("loading");
+    } else if (
+      transactionData?.status === "PENDING" ||
+      transactionData?.status === "ACCEPTED_ON_L2" ||
+      transactionData?.status === "ACCEPTED_ON_L1"
+    ) {
+      setScreen("success");
+    }
+  }, [isError, transactionData]);
+
   useEffect(() => {
     if (account) {
       // Our Indexer
       setLoading(true);
       fetch(
-        `${process.env.NEXT_PUBLIC_SERVER_LINK}/addr_to_full_ids?addr=${hexToDecimal(account.address)}`
+        `${
+          process.env.NEXT_PUBLIC_SERVER_LINK
+        }/addr_to_full_ids?addr=${hexToDecimal(account.address)}`
```

Comment:
IfI'm not mistaken, this should work even without hexToDecimal

Reply:
No it does not